clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -808,7 +808,8 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 }
                 log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
                 this.subscriptions.subscribe(topics, listener);
-                metadata.setTopics(subscriptions.groupSubscription());
+                this.metadata.needMetadataForAllTopics(true);
```

Comment:
@hachikuji Well the reason I removed the line `metadata.setTopics(subscriptions.groupSubscription());` is we make the same call in `addMetadataListener()` as part of the changes made in this PR ([here](https://github.com/vahidhashemian/kafka/blob/2a77e8c11bcc372f67a3dee95aa7f8eafbf1f3ec/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L166)). And I now realize we may not need the new line 811 above because, as I understand, we require fresh metadata for all topics when we are dealing with regex subscription. So, perhaps we just remove the line `this.metadata.needMetadataForAllTopics(true);`?